Add leaderboards

vladh opened the pull request:
This PR adds leaderboards, which are currently only visible in the development environment, i.e. will not show up on the deployed website. The leaderboards currently use some auto-generated dummy data, because we don't have enough members to accurately test them.

The leaderboard lives in the `Leaderboard` component, which has two possible features. With `grouped={false}`, the leaderboard is ungrouped, which means that it simply displays a list of all members sorted by their latest $/dev amount. With `grouped={true}`, the leaderboard is grouped into brackets depending on developer count, which can be defined in `DEV_GROUP_BOUNDS`.

Note that, because annual reports can happen at different times and based on different fiscal years, the amounts listed in the leaderboard for each company might have been declared up to a year apart from each other, so this is a best-effort comparison.

Also, the example avatars I added can be removed at any time in the future, they are 500 bytes each.

Fixes #42.

![2024-08-07-19-41-02](https://github.com/user-attachments/assets/ec31226f-10ec-40a4-8fda-fa0084186ace)

![2024-08-07-19-41-12](https://github.com/user-attachments/assets/728b9c69-300f-4ae9-94ef-cfb5b4216305)

dcramer commented:
I personally don't think a leaderboard is a good idea. Its just going to put the biggest sponsor on top all the time, so whats the motivating purpose?

chadwhitacre commented:
> whats the motivating purpose?

Discovery for new visitors and social validation for members. I can agree that promoting individual competition between members might not be the best framing. We do need some kind of browse interface (also need search). 🤔

dcramer commented:
I think its more so that its not really competition - bigger company will almost always be bigger. Not much of a leaderboard at the end of the day. Doesn't create an incentive if its a page no one will ever show up on.

chadwhitacre commented:
> bigger company will almost always be bigger

Wdym? Isn't $/dev meant to mitigate this?

vladh commented:
Argument for leaderboards: they motivate companies to donate more, perhaps not in individual competition but relative to the average, so as not to be seen to be doing the least.

Argument against leaderboards: those who can afford to give the most are disproportionately visible, and the top 10/20 might change slowly, which makes new members who donate less very difficult to notice, especially as the list grows longer.

Potential solution: show newest members first on the homepage, have a separate page, more or less visible, with the highest-first sorted amounts.

chadwhitacre commented:
We could also present the data more in tabular fashion with grouping and pagination and let users sort and group for themselves. Don't call it a leaderboard but more a tool for data exploration and comparison.

dcramer commented:
I like both of the last ideas:

- Make it more of a company explorer (think the charity explorer thing?)
- Find ways to promote new companies that have joined, or companies that have grown their contributions a bunch

chadwhitacre commented:
For this PR maybe we could just rename to "Company Explorer" and land it? It's all dummy data in dev mode at this point anyway, right @vladh?

vladh commented:
@chadwhitacre I was just doing that actually, haha. I renamed the leaderboard to “Member Company Explorer” and moved it to `/members`, which is not linked from anywhere.

chadwhitacre commented:
Looks good locally. Let's go with just "Member Companies" for the heading now that it's on `/members` (where it should be, good call).

vladh commented:
@chadwhitacre Done!

chadwhitacre commented:
`members/test.json` will show up in prod as well, no? Probably shouldn't?

vladh commented:
@chadwhitacre I had already removed that off of `main`, but I just removed it here too to make sure the merge gets resolved correctly.
